crates/bevy_app/src/plugin_group.rs

```diff
@@ -2,6 +2,123 @@ use crate::{App, AppError, Plugin};
+/// #   #[derive(Default)] pub struct InternalPlugin;
+/// #   impl Plugin for InternalPlugin { fn build(&self, _: &mut App) {} }
+/// # }
+/// plugin_group!(
+///     PhysicsPlugins {
+///         // Due to local ambiguity issues, you have to
+///         // use a colon before the plugin's name.
+///         :TickratePlugin,
+///         // Due to local ambiguity issues, you have to
+///         // use 3 colons for the last part of the path.
+///         collision::capsule:::CapsuleCollisionPlugin,
+///         velocity:::VelocityPlugin,
+///         // Add a documented feature restriction.
+///         #[cfg(feature = "external_forces")]
+///         features:::ForcePlugin,
+///         // You can add any attribute you want like this, but it won't be documented.
+///         #[custom(cfg(target_arch = "wasm32"))]
+///         web:::WebCompatibilityPlugin,
+///         // You can hide plugins from documentation.
+///         // NOTE: Due to macro limitations, you can only use
+///         //       #[doc(hidden)] for plugins at the bottom of the macro.
+///         #[doc(hidden)]
+///         internal:::InternalPlugin
+///     }
+/// );
+/// ```
+#[macro_export]
+macro_rules! plugin_group {
+    (
+        $(#[$group_meta:meta])*
+        $group:ident {
+            $(
+                $(#[cfg(feature = $plugin_feature:literal)])?
+                $(#[custom($plugin_meta:meta)])*
+                $($plugin_path:ident::)* : $plugin_name:ident
+            ),*
+            $(
+                ,$(
+                    #[doc(hidden)]
+                    $(#[cfg(feature = $hidden_plugin_feature:literal)])?
+                    $(#[custom($hidden_plugin_meta:meta)])*
+                    $($hidden_plugin_path:ident::)* : $hidden_plugin_name:ident
+                ),+
+            )?
+
+            $(,)?
+        }
+        $($(#[doc = $post_doc:literal])+)?
+    ) => {
+        $(#[$group_meta])*
+        ///
+        $(#[doc = concat!(
+            "* [`", stringify!($plugin_name), "`](" $(, stringify!($plugin_path), "::")*, stringify!($plugin_name), ")"
+            $(, " - with feature `", $plugin_feature, "`")?
+        )])*
+        $(
+            ///
+            $(#[doc = $post_doc])+
+        )?
+        pub struct $group;
+        impl PluginGroup for $group {
+            fn build(self) -> PluginGroupBuilder {
+                let mut group = PluginGroupBuilder::start::<Self>();
+
+                $(
+                    $(#[cfg(feature = $plugin_feature)])?
+                    $(#[$plugin_meta])*
+                    {
+                        group = group.add(<$($plugin_path::)*$plugin_name>::default());
+                    }
+                )*
+                $($(
+                    $(#[cfg(feature = $hidden_plugin_feature)])?
+                    $(#[$hidden_plugin_meta])*
+                    {
+                        group = group.add(<$($hidden_plugin_path::)*$hidden_plugin_name>::default());
+                    }
```

Comment:
Same thing as my other comment. :)

```suggestion
                    {
                        const _: () = {
                            const fn check_default<T: Default>() {}
                            check_default::<$($hidden_plugin_path::)*$hidden_plugin_name>();
                        };

                        group = group.add(<$($hidden_plugin_path::)*$hidden_plugin_name>::default());
                    }
```